task8/ServletsOnMaven/src/main/java/ServletsOnPosts/LoginServlet.java

```diff
@@ -0,0 +1,43 @@
+package ServletsOnPosts;
+
+import Posts.JSONDecorator;
+import Services.UserService;
+import UtilClasses.User;
+import org.apache.commons.io.IOUtils;
+
+import javax.annotation.Resource;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class LoginServlet extends HttpServlet {
+
+
+    @Resource(name="jdbc/mydb")
+    private DataSource dataSource;
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        PrintWriter writer = resp.getWriter();
+        String body = IOUtils.toString(req.getReader());
+        User postRequest = JSONDecorator.getGson().fromJson(body, User.class);
+        try {
+            UserService.checkForUser(postRequest, dataSource);
+        }
+        catch (UserService.NoUser noUser) {
+            writer.write(JSONDecorator.toJson("No such user"));
+            return;
+        }
+        catch (UserService.DifferencePasswords differencePasswords) {
+            writer.write(JSONDecorator.toJson("Passwords are not equal"));
```

Comment:
Are you reporting that the user-input password does not match the one stored in DB?
If so, this is still a security flaw. Nowadays, most ofter the server replies "No such user of a bad password". We don;t give a spare confirmation that such user exists thus welcoming an intruder to try and brute-froce a password